### ISSUE TITLE
[stacks-coordinator] Stacks node integration

### ISSUE THREAD

**Issue by netrome** [Trust-Machines/core-eng#61]
The Coordinator needs to interact with stacks nodes to

1. Get peg-in and peg-out operations
2. Query the sBTC contract
3. Broadcast stacks transactions

The integration should roughly correspond to the following trait
```rust
pub trait StacksNode {
    fn get_peg_in_ops(&self, block_height: usize) -> Vec<PegInOp>;
    fn get_peg_out_request_ops(&self, block_height: usize) -> Vec<PegOutRequestOp>;
    fn burn_block_height(&self) -> usize;
    fn next_nonce(&self, addr: StacksAddress);
    fn broadcast_transaction(&self, tx: &StacksTransaction);
}
```

## Definition of done
1. The `stacks-coordinator` has an implementation of the `StacksNode` trait which interacts with a stacks node.
2. The interaction is tested with a local instance of a stacks node.

```[tasklist]
### Dependencies
- [ ] https://github.com/stacks-network/stacks-blockchain/issues/3581
```

**Comment by stjepangolemac** [comment]
I can't run stacks-node locally without crashing. This is blocked by https://github.com/stacks-network/stacks-blockchain/issues/3581.

**Comment by stjepangolemac** [comment]
Maybe we don't even need to run mocknet for this one. It seems I can run mainnet locally without issues.

The problem now is that even if I run the stacks-node from the branch in this PR https://github.com/stacks-network/stacks-blockchain/pull/3577 I don't seem to be getting the right data when I ask for `burn_ops`.

```
curl https://stacks-node-api.testnet.stacks.co/v2/burn_ops/2421185/peg_in

/v2/burn_ops/2421185/peg_in% # This is the response
```

Do you have any thoughts @netrome?

**Comment by stjepangolemac** [comment]
I need `StacksTransaction` type with its serialization logic to proceed. Blocked by https://github.com/Trust-Machines/core-eng/issues/14

**Comment by donpdonp** [comment]
Peg-In response from stacks-node
https://github.com/stacks-network/stacks-blockchain/pull/3511

```
{
  "PegIn": [
    {
      "amount": 1337,
      "block_height": 217,
      "burn_header_hash": "7ce0139a3fe356316409fe704dbdb9ac8f1847c168fe68bd913d0df75c359fe1",
      "peg_wallet_address": {
        "Addr32": [
          false,
          "P2TR",
          [ 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0 ]
        ]
      },
      "recipient": "S0000000000000000000002AA028H",
      "recipient_contract_name": null,
      "txid": "84808f747841fb26e74f171487e56eda1a51373581eb04a8c87a421a77bd46d2",
      "vtxindex": 1
    }
  ]
}
```

**Comment by donpdonp** [comment]
the PR for this ticket is about to get merged so whats left on this ticket was put into a new issue #172
